src/ec/eddsa.rs

```diff
@@ -176,10 +176,56 @@ impl signature::VerificationAlgorithm for EdDSAParameters {
         }
         let msg = msg.as_slice_less_safe();
         let signature = signature.as_slice_less_safe();
-        bssl::map_result(unsafe {
-            GFp_ed25519_verify(msg.as_ptr(), msg.len(), signature.as_ptr(),
-                               public_key.as_ptr())
-        })
+
+        let mut a = Point::new();
+        if (signature[63] & 224) != 0 {
+            return Err(error::Unspecified);
+        }
+        let ret = unsafe {
+            GFp_x25519_ge_frombytes_vartime(&mut a, public_key.as_ptr())
+        };
+        if ret != 0 { return Err(error::Unspecified) }
```

Comment:
IMO it makes more sense to change the return value convention of `GFp_x25519_ge_frombytes_vartime()` (in an earlier commit) to match the BoringSSL convention (1 is success, 0 is error) so that we can use `try!(bssl::map_result(...))` here instead.